src/dos/dos_devices.cpp

```diff
@@ -375,6 +375,61 @@ DOS_File &DOS_File::operator=(const DOS_File &orig)
 	return *this;
 }
 
+void DOS_SetMediaDelay(const bool use_media_delay)
+{
+	auto get_delay_percent = []() {
+		LOG_MSG("machine = %d", machine);
+		switch (machine) {
+		case MCH_HERC: return 300;
+		case MCH_CGA:
+		case MCH_TANDY:
+		case MCH_PCJR: return 150;
+		case MCH_EGA: return 125;
+		case MCH_VGA:
+		default: return 100;
+		}
+	};
+
+	DOS_File::MediaDelayGuard::SetDelayPercent(
+	        use_media_delay ? get_delay_percent() : 0);
+
+	LOG_MSG("DOS: Media will %s delays",
+	        use_media_delay ? "use realistic" : "run without");
+}
+
+DOS_File::MediaDelayGuard::~MediaDelayGuard()
+{
+	if (delay_percentage == 0) {
+		return;
+	}
+	auto scaled_ops = (GetNumOperations() * delay_percentage) / 100;
+	while (scaled_ops--) {
+		CALLBACK_Idle();
+	}
+}
+
+constexpr void DOS_File::MediaDelayGuard::SetDelayPercent(const int percentage) noexcept
+{
+	assert(percentage >= 0);
+	delay_percentage = percentage;
+}
+
+constexpr uint16_t DOS_File::MediaDelayGuard::GetNumOperations() const noexcept
+{
+	constexpr uint16_t ReadBytesPerOp  = 128;
+	constexpr uint16_t WriteBytesPerOp = 192;
+	constexpr uint16_t SeekMultiplier  = 20;
```

Comment:
`int`s all the way :)

Reply:
Would you care to elaborate a bit in a comment how you ended up with these magic constants?